layouts/shortcodes/feature-gate-table.html

```diff
@@ -0,0 +1,114 @@
+{{- $featureDataFiles := .Site.GetPage "page" "docs/reference/command-line-tools-reference/feature-gates" -}}
+
+<!-- Check if 'show-removed' is passed to the shortcode to display only 'removed' feature gates -->
+{{- $removedFeatureGateRequested := .Get "show-removed" -}}
+
+<!-- Extract value for the 'include' parameter passed to the shortcode  -->
+{{- $includeValues := .Get "include" -}} 
+
+<!-- Sort Feature gate pages list -->
+{{- $sortedFeatureGates := sort ($featureDataFiles.Resources.ByType "page") -}}
+
+<table class="sortable-table">
+    <caption style="display:none">
+        {{- .Get "caption" -}}
+    </caption>
+    <thead>
+      <tr>
+          <th>{{- T "feature_gate_table_header_feature" -}}</th>
+          <th>{{- T "feature_gate_table_header_default" -}}</th>
+          <th>{{- T "feature_gate_table_header_stage" -}}</th>
+          <th>
+          {{- if $removedFeatureGateRequested -}}
+               {{- T "feature_gate_table_header_from" -}}
+            {{- else -}}
+               {{- T "feature_gate_table_header_since" -}}
+            {{- end -}}
+          </th>
+          <th>
+            {{- if $removedFeatureGateRequested -}}
+                 {{- T "feature_gate_table_header_to" -}}
+              {{- else -}}
+                 {{- T "feature_gate_table_header_until" -}}
+              {{- end -}}
+            </th>
+      </tr>
+    </thead>
+    <tbody>
+    {{- range $featureGateFile := $sortedFeatureGates -}}
+
+        <!-- Extract the feature gate name from the "Title" parameter in file -->
+        {{- $featureGateName := $featureGateFile.Params.Title -}}
+
+        <!-- Extract the feature gate status (i.e. removed or not) from the "Removed" parameter in file -->
+        {{- $removedStatusForFeature := index $featureGateFile.Params.Removed -}}
+
+        <!-- Check if 'stages' parameter is missing in the front matter -->
+        {{- if not (isSet $featureGateFile.Params "stages") -}}
+            {{- warnf "Stages parameter is missing in the front matter for %s in %s" $featureGateName (print $featureGateFile.File.Lang "/" $featureGateFile.File.Path) -}}
```

Comment:
Included this log message here; found it helpful during implementation to identify files lacking stages. Could potentially aid localization team.